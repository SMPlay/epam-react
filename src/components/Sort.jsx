import React from "react";
import { createUseStyles } from "react-jss";

const useStyles = createUseStyles({
  sortSection: {
    background: "rgb(219, 219, 219)",
  },
  sort: {
    display: "flex",
    flexDirection: "row",
    justifyContent: "space-between",
  },
});

const Sort = (props) => {
  const {
    metaText, onClick, sortBy, optionalComponent,
  } = props;
  const classes = useStyles();

  return (
    <section className={classes.sortSection}>
      <div className="container">
        <div className="row">
          <div className={classes.sort}>
            <div>
              <p>{metaText}</p>
            </div>
            {optionalComponent}
          </div>
        </div>
      </div>
    </section>
  );
};

export default Sort;
