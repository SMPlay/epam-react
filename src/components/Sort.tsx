import React, { ReactNode } from "react";
import { createUseStyles } from "react-jss";

const useStyles = createUseStyles({
  sortSection: {
    background: "rgb(219, 219, 219)",
  },
  sort: {
    display: "flex",
    flexDirection: "row",
    justifyContent: "space-between",
  },
});

interface Props {
  metaText: ReactNode;
}

export const Sort: React.FC<Props> = (props) => {
  const {
    metaText, children,
  } = props;
  const classes = useStyles();

  return (
    <section className={classes.sortSection}>
      <div className="container">
        <div className="row">
          <div className={classes.sort}>
            <div>
              <p>{metaText}</p>
            </div>
            {children}
          </div>
        </div>
      </div>
    </section>
  );
};
