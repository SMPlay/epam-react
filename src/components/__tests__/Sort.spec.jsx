import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import { BrowserRouter as Router } from "react-router-dom";
import { Sort } from "../Sort";

describe("Sort component", () => {
  let props;
  beforeEach(() => {
    props = {
      metaText: "meta text",
    };
  });

  it("should render with props", () => {
    const tree = renderer.create(
      <Router>
        <Sort metaText={props.metaText}>
          <div />
        </Sort>
      </Router>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should rendre meta text", () => {
    const component = mount(
      <Router>
        <Sort metaText={props.metaText}>
          <div />
        </Sort>
      </Router>,
    );

    expect(component.find("p").text()).toBe("meta text");
  });
});
