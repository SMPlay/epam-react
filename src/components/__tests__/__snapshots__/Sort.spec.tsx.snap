// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort component should render with props 1`] = `
<section
  className="sortSection-0-2-1"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="sort-0-2-2"
      >
        <div>
          <p>
            meta text
          </p>
        </div>
        <div />
      </div>
    </div>
  </div>
</section>
`;
